File: P1/E1/FactoriaCarretera.java
package E1;

public class FactoriaCarretera implements FactoriaCarreraYBicicleta {

    @Override
    public Bicicleta CrearBicicleta(int id){
        return new BicicletaCarretera(id);
    }

    @Override
    public Carrera CrearCarrera(){
        return new CarreraCarretera();
    }
}

File: P1/E2/FactoriaCarretera.py
from BicicletaCarretera import BicicletaCarretera
from CarreraCarretera import CarreraCarretera
from FactoriaCarreraYBicicleta import FactoriaCarreraYBicicleta

class FactoriaCarretera(FactoriaCarreraYBicicleta):
    def crearBicicleta(self, id: int):
        return BicicletaCarretera(id)

    def crearCarrera(self):
        return CarreraCarretera()
